tensorboard/uploader/exporter_test.py

```diff
@@ -227,6 +229,328 @@ def outdir_files():
             },
         )
 
+    def test_e2e_success_case_with_finished_blob_sequence_data(self):
```

Comment:
Can we try to have just one end-to-end test case that exercises as much
behavior as possible? It looks like we should be able to get all happy
paths: scalars, finished blob sequences, unfinished blob sequences, blob
sequences with holes.

End-to-end tests like this are great for coverage, but require a ton of
code for setup and mock implementations, so it’s much easier to read
(and debug failures) if we can do that only once on a carefully designed
input that covers everything. Once the end-to-end test convinces us that
the main wiring is correct, the rest can be unit-tested. For example,
note how comparatively simple all the error case `test_*` methods are,
spanning about 20 lines instead of about 100–200.